src/include/ipc4/notification.h

```diff
@@ -42,6 +42,68 @@ enum sof_ipc4_notification_type {
 	SOF_IPC4_MANAGEMENT_SERVICE		= 16,
 };
 
+/**
+ * \brief Resource Event Notification provides unified structure for events that may be raised by
+ * identifiable entity from inside the FW (like a module instance identifiable by module instance
+ * ID, or a pipeline identifiable by pipeline ID).
+ */
+enum sof_ipc4_resource_event_type {
+	/* MCPS budget for the module exceeded */
+	SOF_IPC4_BUDGET_VIOLATION		= 0,
+	/* Underrun detected by the Mixer */
+	SOF_IPC4_MIXER_UNDERRUN_DETECTED	= 1,
+	/* Stream data segment completed by the gateway */
+	SOF_IPC4_STREAM_DATA_SEGMENT		= 2,
+	/* Error caught during data processing */
+	SOF_IPC4_PROCESS_DATA_ERROR		= 3,
+	/* Stack overflow in a module instance */
+	SOF_IPC4_STACK_OVERFLOW			= 4,
+	/* KPB changed buffering mode */
+	SOF_IPC4_BUFFERING_MODE_CHANGED		= 5,
+	/* Underrun detected by gateway. */
+	SOF_IPC4_GATEWAY_UNDERRUN_DETECTED	= 6,
+	/* Overrun detected by gateway */
+	SOF_IPC4_GATEWAY_OVERRUN_DETECTED	= 7,
+	/* DP task missing the deadline */
+	SOF_IPC4_EDF_DOMAIN_UNSTABLE		= 8,
+	/* Watchdog notification */
+	SOF_IPC4_WATCHDOG_EVENT			= 9,
+	/* IPC gateway reached high threshold */
+	SOF_IPC4_GATEWAY_HIGH_THRES		= 10,
+	/* IPC gateway reached low threshold */
+	SOF_IPC4_GATEWAY_LOW_THRES		= 11,
+	/* Bit Count Error detected on I2S port */
+	SOF_IPC4_I2S_BCE_DETECTED		= 12,
+	/* Clock detected/loss on I2S port */
+	SOF_IPC4_I2S_CLK_STATE_CHANGED		= 13,
+	/* I2S Sink started/stopped dropping data in non-blk mode */
+	SOF_IPC4_I2S_SINK_MODE_CHANGED		= 14,
+	/* I2S Source started/stopped generating 0's in non-blk mode */
+	SOF_IPC4_I2S_SOURCE_MODE_CHANGED	= 15,
+	/* Frequency drift exceeded limit in SRE */
+	SOF_IPC4_SRE_DRIFT_TOO_HIGH		= 16,
+	/* The notification should be sent only once after exceeding threshold or aging timer */
+	SOF_IPC4_TELEMETRY_DATA_STATUS		= 17,
+	/* VAD event detected */
```

Comment:
Could you change this comment to some more "generic"? Like: "SNDW debug notification e.g. external VAD detected"